storage/replica.go

```diff
@@ -1660,3 +1677,49 @@ func (r *Replica) maybeAddToSplitQueue() {
 		r.store.splitQueue.MaybeAdd(r, r.store.Clock().Now())
 	}
 }
+
+// processReplicaCmd get a raft command from command queue and
+// call processRaftCommand to execute in a loop until there is
+// not any command in the queue for 5 seconds.
+func (r *Replica) processReplicaCmd() {
+	for {
+		select {
+		case cmd := <-r.applyQueueChan:
+			err := r.processRaftCommand(cmd.idKey, cmd.index, cmd.raftCmd)
+			if cmd.callback != nil {
+				cmd.callback(err)
+			}
+
+		case <-time.After(5 * time.Second):
```

Comment:
Goroutines are cheap to start and stop; does it make sense to keep the goroutine alive for 5 seconds instead of letting it stop in a `default` as soon as the channel becomes empty? I think this may be cheaper than creating a `time.After` for every command.